cmplibrary/src/main/res/raw/js_receiver.js

```diff
@@ -17,7 +17,8 @@ function consentData(event) {
 };
 
 function isFromPM(event) {
-    return !!event.data.payload;
+    event.data.settings = event.data.settings || {};
```

Comment:
here you don't have to change the original obj. Instead you can try a inline check like this:
```Javascript
return event.data.fromPM || event.data.settings && event.data.settings.vendorList;
```

Reply:
Yes we can do this , also we need to have null check for vendorList otherwise this will return false always.